### PR TITLE
[V0 Deprecation] Remove code related to per-request logits processors

### PR DESCRIPTION
**Pull request by DarkLight1337**
## Purpose

Removed the following as they're not valid in V1:
- `ModelConfig.logits_processor_pattern`
- `SamplingParams.logits_processors`
- `ChatCompletionRequest.logits_processor`
- `CompletionRequest.logits_processor`

cc @afeldman-nm 

## Test Result

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [ ] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [ ] The test plan, such as providing test command.
- [ ] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
- [ ] (Optional) Release notes update. If your change is user facing, please update the release notes draft in the [Google Doc](https://docs.google.com/document/d/1YyVqrgX4gHTtrstbq8oWUImOyPCKSGnJ7xtTpmXzlRs/edit?tab=t.0).
</details>

### CODE REVIEW

File: vllm/engine/arg_utils.py [hunk 1]
```diff
@@ -1320,13 +1315,11 @@ def create_model_config(self) -> ModelConfig:
             mm_encoder_tp_mode=self.mm_encoder_tp_mode,
             mm_encoder_attn_backend=self.mm_encoder_attn_backend,
             pooler_config=self.pooler_config,
-            logits_processor_pattern=self.logits_processor_pattern,
             generation_config=self.generation_config,
             override_generation_config=self.override_generation_config,
             enable_sleep_mode=self.enable_sleep_mode,
             model_impl=self.model_impl,
             override_attention_dtype=self.override_attention_dtype,
```
[the remark this answers is not in the document]
Comment (DarkLight1337) [reply vllm/engine/arg_utils.py:1322]: Fixed

**Review by WoosukKwon — APPROVED** [review]
